tests/unit/utils/Utils.cpp

```diff
@@ -68,3 +70,18 @@ char * mktemp_cst(char * __template)
 
     return __template;
 }
+
+char* generate_file_name(char* file_name)
```

Comment:
so basically:
```
void generate_temp_filename(char* filename)
```

after you call `generate_temp_filename` then the passed `filename` will contain the right path :)